src/useAsyncRetry.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback, useState } from 'react';
+import useAsync from './useAsync';
+
+const useAsyncRetry = <T>(fn: () => Promise<T>, args: any[] = []) => {
+  const [attempt, setAttempt] = useState<number>(0);
+  const memoized = useCallback(async () => await fn(), [...args, attempt]);
```

Comment:
Can this be written without `async/await` an extra function wrapping?

```js
const memoized = useCallback(fn, [...args, attempt]);
```

Reply:
I tried this first (it seemed more intuitive to me) but for some reason it wouldn't work on the storybook story, due to the original async function having already resolved. Wrapping it in another async seemed to make it work as advertised.

Reply:
I actually don't think there's a need to memoise the `fn` with `useCallback` if there's no need to manipulate it (since it's a function it will always be passed by reference anyway). I think by creating an async factory wrapper around it, it's probably more likely to have a new async state since we're generating a new async function (and therefore will need memoisation with `useCallback`).

Reply:
Now that I think about it even further, it could even just be simplified to `useCallback(() => fn(), [...args, attempt])` since the return value of `fn` will always be a promise anyway.